compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/AbstractLexer.java

```diff
@@ -284,8 +284,8 @@ protected static boolean isUnicodePatternWhiteSpaceChar(int c) {
      * <code>NumericEscape := \ u { CodePoint }</code>
      */
     protected void processNumericEscape() {
```

Comment:
Shall we split this into two methods?
```java
protected void processNumericEscape() {
    reader.advance(); // Advance for '\'
    processNumericEscapeWithoutBackslash();
}

protected void processNumericEscapeWithoutBackslash() {
    reader.advance(2); // Advance for 'u {'
    ...
}
```

Reply:
In the `processNumericEscape()` call in DocumentationLexer we want to increment the reader for the backslash as well. Seems with the new logic here, it is not happening now.